config/registry.go

```diff
@@ -11,6 +12,26 @@ var (
 	options     = make(map[string]*Option)
 )
 
+// ForEachOption calls fn for each defined option. If fn returns
+// and error the iteration is stopped and the error is returned.
+// Note that ForEachOption does not guarantee a stable order of
+// iteration between multiple calles. ForEachOption does NOT lock
+// opt when calling fn.
+func ForEachOption(fn func(opt *Option) error) error {
+	optionsLock.Lock()
+	defer optionsLock.Unlock()
+
+	for _, opt := range options {
+		opt.Lock()
+		if err := fn(opt); err != nil {
+			opt.Unlock()
+			return err
+		}
+		opt.Unlock()
```

Comment:
`ForEachOption does NOT lock opt when calling fn.` 🤔

Reply:
^^ that was a merge conflict and I've overseen the locking part :D Will fix it